src/System.Net.WebSockets.WebSocketProtocol/src/System/Net/WebSockets/ManagedWebSocketExtensions.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.WebSockets
+{
+    internal static class ManagedWebSocketExtensions
+    {
+        internal static string GetString(this UTF8Encoding encoding, Span<byte> bytes)
+        {
+            return encoding.GetString(bytes.ToArray());
```

Comment:
This is super expensive.  It should use the `GetString` overload that takes `byte*`/`int` and use GetReference and fixed to get such a pointer.